wcfsetup/install/files/lib/system/form/builder/field/IPatternFormField.class.php

```diff
@@ -0,0 +1,28 @@
+<?php
+namespace wcf\system\form\builder\field;
+
+/**
+ * Represents a form field that supports the `pattern` attribute.
+ *
+ * @author      Matthias Schmidt
+ * @copyright   2001-2020 WoltLab GmbH
+ * @license     GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package     WoltLabSuite\Core\System\Form\Builder\Field
+ * @see         https://html.spec.whatwg.org/multipage/input.html#attr-input-pattern
+ * @since       5.4
+ */
+interface IPatternFormField {
+	/**
+	 * Returns the `pattern` attribute of the form field.
+	 *
+	 * If `null` is returned, no `pattern` attribute will be set.
+	 */
+	public function getPattern(): ?string;
+	
+	/**
+	 * Sets the `pattern` attribute of the form field.
+	 *
+	 * If `null` is given, the attribute is unset.
```

Comment:
Maybe add a remark that the pattern will *not* be validated automatically.

Reply:
The form builder API generally documents whenever an exception can be expected due to validation so that the lack thereof for this method already implies no such validation takes place.